src/uint/boxed.rs

```diff
@@ -245,6 +258,13 @@ impl BoxedUint {
 }
 
 impl NonZero<BoxedUint> {
+    /// TODO: this is not really "const", but I need a way to return (value, choice) since
+    /// BoxedUint is not [`ConditionallySelectable`] so `CtChoice::map` and such does not work
+    pub fn const_new(n: BoxedUint) -> (Self, Choice) {
+        let nonzero = n.is_nonzero();
+        (Self(n), nonzero)
+    }
```

Comment:
I factored these into methods on e.g. in this case `BoxedUint`: https://github.com/RustCrypto/crypto-bigint/pull/484

You can use `ConstCtChoice` now.